server/src/main/java/org/elasticsearch/cluster/SnapshotsInProgress.java

```diff
@@ -281,13 +294,20 @@ public String nodeId() {
             return nodeId;
         }
 
+        public String generation() {
```

Comment:
Tracking the generation here as well as via the `String` callback to have it available for retries of shard status updates on e.g. master failover in `SnapshotShardsService`.